WooCommerce: Setup/NUX: Add manage fledgling UX

allendav opened the pull request:
Work in progress

See #13613

#### How it Works
- Depending on whether the merchant has completed setup or not, we will either render the "Howdy" (egg) or the "Now that your store setup is complete." (but no orders yet) (fledgling) view
- Those containers then render the widgets for each of those views
- There is a intermediate container, widget-group, that dynamically handles rendering full width, half-width, or third-width children using flexbox tricks
- This PR includes the parts of the momma view shared with fledgling too

#### To test:

- First, npm run test-client client/extensions/woocommerce/state
- Navigate to http://calypso.localhost:3000/store/:site
- Make sure you see the “Howdy! Let's set up your store & start selling” view aka the egg view
- Note: The “I’m finished” button in the egg view is not hooked up yet (next PR)
- Use the https://developer.wordpress.com/docs/api/console/ to POST to /sites/{siteID}/calypso-preferences/woocommerce setting finished_initial_setup to 1
- Refresh the page
- Make sure you see the “Now that your setup is complete it\'s time to spread the word” view aka the fledgling view
- Make sure each sharing link works in the share widget
- Make sure you see the three latest articles from the “Start your store” category on woocommerce.com in the reading widget
- Make sure clicking in the email address field expands the footer of the reading widget
- Make sure the subscribe button is not enabled until a valid email address and at least one role is picked
- Note: The “subscribe” button in the reading widget is not hooked up yet (next PR)
- Make sure the example order widget appears
- Note: There is no example order to display yet, so clicking that button doesn’t do much good yet
- Make sure the test and view my site widget appears
- Make sure clicking on the view my site widget button opens a new tab to your site
- Lastly, to see the new components in the momma view, force `storeHasOrders` to true in `render` in woocommerce/app/dashboard/manage.js

cc @kellychoffman @jameskoster

a8ci18n commented on client/extensions/woocommerce/app/dashboard/manage-no-orders.js:
```diff
@@ -0,0 +1,138 @@
+		return (
+			<BasicWidget
+				className="dashboard__theme-widget"
+				buttonLabel={ translate( 'View themes' ) }
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'View Themes' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

kellychoffman commented:
👍 This looks good to me.

a8ci18n commented on client/extensions/woocommerce/app/dashboard/manage-no-orders.js:
```diff
@@ -0,0 +1,116 @@
+				text={ translate( 'Choice articles for new store owners to read. Get tips' +
+					' from seasoned merchants, learn best practices to keep your store' +
+					' ship-shape and prepare for success.' ) }
+				title={ translate( 'Recommended reading' ) }
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Recommended Reading' )` *ES Score: 22*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-dashboard-fledgling)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

belcherj commented on client/extensions/woocommerce/state/sites/setup-choices/selectors.js:
```diff
@@ -0,0 +1,50 @@
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { LOADING } from 'woocommerce/state/constants';
+
+const getRawSetupChoices = ( state, siteId ) => {
```
why not just getSetupChoices? Danny used getApiSetupChoices in shipping

allendav replied:
Agreed! I was copying what we did in settings, but this is simple enough. Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/fc929a9089c59738059bec4d882378d93712696e

kellychoffman commented:
I updated the copy per @MarinaWoo's review. I didn't see where we are pulling in the links from in the reading widget. The request is to add two to the end and change the order of the second two. So it would be: 

> Four ways to make managing your store's updates easier https://woocommerce.com/2017/05/manage-woocommerce-updates-easily/ 
> May 19, 2017
>  
> 10 powerful extensions for new WooCommerce stores 
https://woocommerce.com/2017/03/10-new-store-extensions/ 
> Mar 6, 2017
>  
> Must-have extensions for selling digital goods 
> https://woocommerce.com/2017/04/extensions-digital-goods/ 
> Apr 7, 2017
>
> How Cured creates personal connections between customers and food
> https://woocommerce.com/2017/03/cured-customer-story/
> Mar 22, 2017
>  
> How to spend your first $50 on marketing
> https://woocommerce.com/2016/07/spend-first-50-marketing/
> Jul 6, 2016

a8ci18n commented on client/extensions/woocommerce/app/dashboard/manage-no-orders.js:
```diff
@@ -0,0 +1,108 @@
+				text={ translate( 'You’re not alone! Get tips from seasoned merchants,' +
+					' learn best practices to keep your store ship-shape,' +
+					' and find how to boost your sales and drive traffic.' ) }
+				title={ translate( 'Recommended reading' ) }
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Recommended Reading' )` *ES Score: 22*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-dashboard-fledgling)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

justinshreve commented on client/extensions/woocommerce/components/widget-group/index.js:
```diff
@@ -0,0 +1,83 @@
+		);
+	}
+
+	renderChildren = () => {
```
Fun! Took me a second to figure out why it was done this way -- so a comment saying how this will split children into correct rows with the correct number per would probably be helpful for future devs that might look in here.

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

justinshreve commented on client/extensions/woocommerce/components/widget-group/index.js:
```diff
@@ -0,0 +1,83 @@
+	}
+}
+
+WidgetGroup.defaultProps = {
```
Since this is a class, this can be defined under `propTypes` above.

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

justinshreve commented on client/extensions/woocommerce/state/action-types.js:
```diff
@@ -13,6 +13,8 @@ export const WOOCOMMERCE_PRODUCT_CREATE = 'WOOCOMMERCE_PRODUCT_CREATE';
 export const WOOCOMMERCE_PRODUCT_CREATE_SUCCESS = 'WOOCOMMERCE_PRODUCT_CREATE_SUCCESS';
 export const WOOCOMMERCE_PRODUCT_EDIT = 'WOOCOMMERCE_PRODUCT_EDIT';
 export const WOOCOMMERCE_PRODUCT_VARIATION_EDIT = 'WOOCOMMERCE_PRODUCT_VARIATION_EDIT';
+export const WOOCOMMERCE_SETUP_CHOICES_REQUEST = 'WOOCOMMERCE_SETUP_CHOICES_REQUEST';
```
These should be under `SETTINGS_...` to keep in alphabetical order

allendav replied:
LOL. I tried. English is hard. Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

justinshreve commented on client/extensions/woocommerce/state/sites/setup-choices/actions.js:
```diff
@@ -0,0 +1,42 @@
+ */
+import {
+	areSetupChoicesLoaded,
+	areSetupChoicesLoading
```
Trailing comma.

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

justinshreve commented on client/extensions/woocommerce/state/sites/setup-choices/actions.js:
```diff
@@ -0,0 +1,42 @@
+import { setError } from '../status/wc-api/actions';
+import {
+	WOOCOMMERCE_SETUP_CHOICES_REQUEST,
+	WOOCOMMERCE_SETUP_CHOICES_REQUEST_SUCCESS
```
Trailing comma.

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

justinshreve commented on client/extensions/woocommerce/state/sites/setup-choices/test/selectors.js:
```diff
@@ -0,0 +1,106 @@
+			expect( areSetupChoicesLoaded( loadedState, 123 ) ).to.be.true;
+		} );
+
+		it( 'should be false when settings are loaded only for a different site.', () => {
```
`settings` -> `setup choices`

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

justinshreve commented on client/my-sites/sidebar/sidebar.jsx:
```diff
@@ -327,7 +327,7 @@ export class MySitesSidebar extends Component {
 			showStoreLink &&
 			<SidebarItem
-				label={ translate( 'Store' ) }
+				label={ translate( 'Store (BETA)' ) }
```
Did we decide to show beta next to this for v1?

allendav replied:
Yep - cc @kellychoffman

allendav commented:
@kellychoffman  wrote:

> I updated the copy per @MarinaWoo's review. I didn't see where we are pulling in the links from in the reading widget. The request is to add two to the end and change the order of the second two. So it would be...

The links are the most recent 3 posts from the "Start your store" category on woocommerce.com

I would strongly recommend NOT hardcoding links to particular posts since the links will rapidly become stale. @avivapinchas suggested the "Start your store" category https://github.com/Automattic/wp-calypso/issues/13613#issuecomment-302465994 - could we continue to use that please?

kellychoffman commented:
> The links are the most recent 3 posts from the "Start your store" category on woocommerce.com

> I would strongly recommend NOT hardcoding links to particular posts since the links will rapidly become stale. 

That makes total sense. Carry on! :)

allendav commented on client/extensions/woocommerce/components/reading-widget/index.js:
```diff
@@ -0,0 +1,197 @@
+						' news delivered to your inbox' ) }
+				</p>
+				{
+					expanded && ( <p>
```
I think I'd like to decline this suggestion - I think `renderSubscriptionFormExplanation` reads well as is

allendav commented on client/extensions/woocommerce/app/dashboard/index.js:
```diff
@@ -20,55 +23,68 @@ class Dashboard extends Component {
 
-	renderStoreManagement = () => {
-		const { translate } = this.props;
+		const newSiteID = newProps.selectedSite && newProps.selectedSite.ID || null;
```
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/fc929a9089c59738059bec4d882378d93712696e

allendav commented on client/extensions/woocommerce/app/dashboard/manage-orders.js:
```diff
@@ -0,0 +1,149 @@
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import { getLink } from '../../lib/nav-utils';
```
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/fc929a9089c59738059bec4d882378d93712696e

allendav commented on client/extensions/woocommerce/components/reading-widget/index.js:
```diff
@@ -0,0 +1,197 @@
+					/>
+				) }
+				{ expanded && (
+					<Button
```
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/fc929a9089c59738059bec4d882378d93712696e

allendav commented:
@justinshreve wrote:

> All of your testing directions work except the last step. When I set storeHasOrders to true, I get a JS error and a blank screen

Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5d5ea99323f57549f169d0a501b10e31ba869d92 - I also deleted all the not-yet-implemented widgets

a8ci18n commented on client/extensions/woocommerce/app/dashboard/manage-orders.js:
```diff
@@ -0,0 +1,80 @@
+				text={ translate( 'You’re not alone! Get tips from seasoned merchants,' +
+					' learn best practices to keep your store ship-shape,' +
+					' and find how to boost your sales and drive traffic.' ) }
+				title={ translate( 'Recommended reading' ) }
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Recommended Reading' )` *ES Score: 22*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-dashboard-fledgling)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

allendav commented on client/extensions/woocommerce/components/reading-widget/index.js:
```diff
@@ -0,0 +1,197 @@
+}
+
+export default connect( ( state, ownProps ) => {
+	const siteId = ownProps.siteId ? ownProps.siteId : ReadingWidget.defaultProps.siteId;
```
Not yet, but since we will probably want to be able to set a different query for "momma" than fledgling (e.g. a different category) it made sense to me to allow setting the blog ID as well.

allendav commented on client/extensions/woocommerce/components/share-widget/index.js:
```diff
@@ -0,0 +1,129 @@
+					pathname: '/sharer.php',
+					query: {
+						u: urlToShare,
+						app_id: config( 'facebook_api_key' ),
```
I chose facebook_api_key based on this existing implementation - https://github.com/Automattic/wp-calypso/blob/a70587157b33783ba4d5e0e4a1e136680d459d51/client/blocks/reader-share/index.jsx#L56

allendav commented on client/extensions/woocommerce/app/dashboard/index.js:
```diff
@@ -20,55 +23,68 @@ class Dashboard extends Component {
-	// TODO - figure out from state if setup has been completed for this store yet
+	const finishedInitialSetup = getFinishedInitialSetup( state );
+	const loading = areSetupChoicesLoading( state );
+	const selectedSite = getSelectedSite( state );
```
Thank you! Good catch. Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/app/dashboard/manage-no-orders.js:
```diff
@@ -0,0 +1,108 @@
+/**
+ * Internal dependencies
+ */
+import BasicWidget from '../../components/basic-widget';
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/app/dashboard/manage-no-orders.js:
```diff
@@ -0,0 +1,108 @@
+		return (
+			<ReadingWidget
+				className="dashboard__reading-widget"
+				site={ site }
```
Thank you! Good catch! Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/app/dashboard/manage-orders.js:
```diff
@@ -0,0 +1,149 @@
+			<ShareWidget
+				text={ translate( 'Share a link to your store on social media.' ) }
+				title={ translate( 'Share your store' ) }
+				urlToShare={ getLink( 'https://:site', site ) }
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/components/basic-widget/index.js:
```diff
@@ -0,0 +1,41 @@
+import Button from 'components/button';
+
+const BasicWidget = ( { buttonLabel, buttonLink, children, className, title } ) => {
+	const classes = classNames( { 'basic-widget__container': true }, className );
```
Yep, fixed a few places - https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/components/reading-widget/index.js:
```diff
@@ -0,0 +1,197 @@
+import emailValidator from 'email-validator';
+import { localize } from 'i18n-calypso';
+import React, { Component, PropTypes } from 'react';
+import ExternalLink from 'components/external-link';
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/components/reading-widget/index.js:
```diff
@@ -0,0 +1,197 @@
+
+	static defaultProps = {
+		query: { category: 'start-your-store', number: 3 },
+		siteId: 113771570,
```
Agreed. Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/components/widget-group/index.js:
```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import React, { Component, PropTypes } from 'react';
+
+/**
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented on client/extensions/woocommerce/components/widget-group/index.js:
```diff
@@ -0,0 +1,83 @@
+	render = () => {
+		const { children, className, title } = this.props;
+		const classes = classNames( {
+			'widget-group__group-container': true,
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc

allendav commented:
@justinshreve @belcherj - thank you for your reviews - all feedback is in and this is ready for another pass - https://github.com/Automattic/wp-calypso/pull/14339

a8ci18n commented on client/extensions/woocommerce/app/dashboard/manage-orders.js:
```diff
@@ -0,0 +1,79 @@
+				text={ translate( 'You’re not alone! Get tips from seasoned merchants,' +
+					' learn best practices to keep your store ship-shape,' +
+					' and find how to boost your sales and drive traffic.' ) }
+				title={ translate( 'Recommended reading' ) }
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Recommended Reading' )` *ES Score: 22*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-dashboard-fledgling)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

allendav commented on client/extensions/woocommerce/app/dashboard/manage-no-orders.js:
```diff
@@ -0,0 +1,106 @@
+class ManageNoOrders extends Component {
+	static propTypes = {
+		site: PropTypes.shape( {
+			slug: PropTypes.string.isRequired,
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/d53efa5922db1e46368e0085ee422ed80acfa7de

allendav commented on client/extensions/woocommerce/app/dashboard/manage-orders.js:
```diff
@@ -0,0 +1,79 @@
+class ManageOrders extends Component {
+	static propTypes = {
+		site: PropTypes.shape( {
+			slug: PropTypes.string.isRequired,
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/d53efa5922db1e46368e0085ee422ed80acfa7de

allendav commented on client/extensions/woocommerce/components/share-widget/index.js:
```diff
@@ -0,0 +1,129 @@
+/**
+ * Internal dependencies
+ */
+import SocialLogo from 'social-logos';
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/d53efa5922db1e46368e0085ee422ed80acfa7de

a8ci18n commented on client/extensions/woocommerce/app/dashboard/manage-orders.js:
```diff
@@ -0,0 +1,80 @@
+				text={ translate( 'You’re not alone! Get tips from seasoned merchants,' +
+					' learn best practices to keep your store ship-shape,' +
+					' and find how to boost your sales and drive traffic.' ) }
+				title={ translate( 'Recommended reading' ) }
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Recommended Reading' )` *ES Score: 22*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-dashboard-fledgling)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).
